### ISSUE TITLE
Extended Ensemble support

### ISSUE THREAD

**Issue by Cortys** [Cortys/SEP-Probly-WS2526#23]
Probly currently only offers a simple full ensemble transformation. This epic adds support for other types of ensembles.

---

## Motivation and Context

Full ensembles can significantly improve model accuracy and are an important tool for uncertainty quantification, but they are often computationally expensive — requiring multiple copies of the entire model to be trained and stored.

Additionally, Ensemble models are currently just (module) lists of models. They are not callable and do not provide any other methods/features.

---

### 1. SubEnsemble support

SubEnsembles are a lightweight alternative to traditional ensembles, where all ensemble members share the same frozen backbone but have independent trainable heads (a smaller set of layers on top). This enables the benefits of ensembling for large-scale models that may otherwise not be feasible.

- Implement a SubEnsemble module that supports shared backbones with multiple independent heads.
- Provide an API similar to the existing ensemble transformation. Since a subensemble is a generalization of a regular ensemble, you could implement the subensemble and then adapt the ensemble transformation to make use of the subensemble transformations.
- Start with a common and torch implementation, but make it so that it is easily extensible to other frameworks such as Flax or TensorFlow.
- Each ensemble head should be independently parameterized and trainable, but share the same input features. 
- Add examples demonstrating the SubEnsemble functionality, training workflow, and performance comparison to standard ensembles.
- Add tests for all of the functionality you implement.
- The backbone must remain fixed and identical across all ensemble members.  

### 2. Support for sklearn ensembles

Scikit-learn already supports [ensemble models](https://scikit-learn.org/stable/api/sklearn.ensemble.html).
For example, a [`RandomForestClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html#sklearn.ensemble.RandomForestClassifier) is an ensemble of [`DecisionTreeClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html#sklearn.tree.DecisionTreeClassifier) models.
If probly is given a sklearn model for which there already is a corresponding ensemble type (DecisionTreeClassifier -> RandomForestClassifier, DecisionTreeRegressor -> RandomForestRegressor).

### 3. Support for mixed ensembles

The members of an ensemble do not have to be copies of each other, nor do they have to be implemented in the same framework.
Probly should offer a MixedEnsemble implementation which accepts a list of arbitrary predictors from which one can sample.

---

## Constraints

- The implementation should be easily extensible to other machine learning frameworks.
- Follow project standards for testing, code style, and reproducibility.  

---
